### DISCUSSION TITLE
Trigger logs are not going to stdout after migrating to 2.6.3 from 2.5.3

### DISCUSSION THREAD

**Discussion by llamageddon83** [apache/airflow#34323] [General] [answered]
Hello Airflow community,

In our triggers, we used a logger like this:

```python
log = logging.getLogger('airflow.task')
```

The logs were going to stdout. We recently upgraded from 2.5.3 to 2.6.3 and logs are no longer going to stdout. Though they are now being displayed in the UI which wasn't the case before. 

Which logger should be used in triggers to get the logs on stdout?

Thank you all for your help.

**Comment by potiuk** [comment]
Getting logs from triggerer to stdout is bad idea because writing them to the stdout is a synchronous operation. You should avoid using any synchronous operations in the async event loop, because it can potentially block the loop and process the events from it.  And that was one of the big changes in 2.6 to properly handle the logs in triggerer. In 2.6, they are written to memory, and then there is  a separate thread that reads the logs from the handler and writes them to the right  task handler (either to the handler that exposes them in the webserver UI or to a remote handler - such as GCS/Stackdriver/etc.) 

You can take two approaches:

1) use one of the remote task handlers we have and use remote logging
2) if you want you can look at the implementation and write your own task handler that will take the logs from memory and print them to stdout in a separate thread (but I would discourage it - using dedicated, remote logging makes a lot more sense.

cc: @dstandish = maybe there is already a handler you could use for that?

**Reply by dstandish** [reply, marked as answer]
i think easiest thing to do is disable indiv trigger logging config:

```
DISABLE_WRAPPER = conf.getboolean("logging", "disable_trigger_handler_wrapper", fallback=False)
```

alternatively you can disable the individual trigger logging behavior by setting both of these attributes to `False` on your handler:

`trigger_should_wrap` and `trigger_supported`

then the individual trigger logging will not be enabled

now, to keep both indiv trigger logging and stdout, well...

by default, when a suitable handler is configured, as you've observed, individual trigger log messages get filtered out by the root handler, which is why they no longer go to stdout.  

the way we achieve this is by adding this filter, which looks for those messages and drops them:

```
    def filter_trigger_logs_from_other_root_handlers(new_hdlr):
        # we add context vars to log records emitted for individual triggerer logging
        # we want these records to be processed by our special trigger handler wrapper
        # but not by any other handlers, so we filter out these messages from
        # other handlers by adding DropTriggerLogsFilter
        # we could consider only adding this filter to the default console logger
        # so as to leave other custom handlers alone
        for h in root_logger.handlers:
            if h is not new_hdlr:
                h.addFilter(DropTriggerLogsFilter())
```

it is conceivable that we could add a config item to not add that filter.  i think you could PR that pretty easily

**Reply by Taragolis** [reply]
> i think easiest thing to do is disable indiv trigger logging config:
DISABLE_WRAPPER = conf.getboolean("logging", "disable_trigger_handler_wrapper", fallback=False)

I've just found that we do not have this configuration option in documentation

**Comment by llamageddon83** [comment]
Thank you @potiuk and @Taragolis for your continued help. 

I will give these a try. I certainly didn't think about stdout being a synchronous operation. 🙏
